go-controller/pkg/clustermanager/routeadvertisements/controller.go

```diff
@@ -482,22 +488,28 @@ func (c *Controller) generateFRRConfigurations(ra *ratypes.RouteAdvertisements)
 	var eipsByNodesByNetworks map[string]map[string]sets.Set[string]
 	getEgressIPsByNode := func(nodeName string) (map[string]sets.Set[string], error) {
 		if eipsByNodesByNetworks == nil {
-			eipsByNodesByNetworks, err = c.getEgressIPsByNodesByNetworks(networkSet)
+			eipsByNodesByNetworks, err = c.getEgressIPsByNodesByNetworks(networks)
 			if err != nil {
 				return nil, err
 			}
 		}
 		return eipsByNodesByNetworks[nodeName], nil
 	}
 
-	// helper to gather host subnets and egress ips as prefixes
-	getPrefixes := func(nodeName string, network string) ([]string, error) {
-		// gather host subnets
+	// helper to gather subnets and egress ips as prefixes
+	getPrefixes := func(nodeName string, network util.NetInfo) ([]string, error) {
```

Comment:
Why not use `selectedNetworks.networkSubnets` instead of NetInfo?

Reply:
I need the topology, the alternative would be to add yet another map for that to selectedNetworks, is that something you prefer to using netInfo?

Reply:
Or are you saying that the logic of determining which subnet to use should happen before calling getPrefixes?

Reply:
Right. I think that is what makes sense, `selectedNetworks` is meant to contain all the pre-processed information in a way that its easy to consume. I would change it to keep the same strategy going on.